includes/reader-revenue/woocommerce/class-woocommerce-cli.php

```diff
@@ -162,6 +162,133 @@ function( $subscription ) {
 		}
 	}
 
+	/**
+	 * Fixes or reports active subscriptions that have missed next payment dates.
+	 * By default, will only process subscriptions started in the past 90 days.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--dry-run]
+	 * : If set, will report results but will not make any changes.
+	 *
+	 * [--batch-size=<batch-size>]
+	 * : The number of subscriptions to process in each batch. Default: 50.
+	 *
+	 * [--start-date=<date-string>]
+	 * : A date string in YYYY-MM-DD format to use as the start date for the script. Default: 90 days ago.
+	 *
+	 * @param array $args Positional args.
+	 * @param array $assoc_args Associative args.
+	 */
+	public function fix_missing_next_payment_dates( $args, $assoc_args ) {
+		$dry_run    = ! empty( $assoc_args['dry-run'] );
+		$now        = time();
+		$batch_size = ! empty( $assoc_args['batch-size'] ) ? intval( $assoc_args['batch-size'] ) : 50;
+		$start_date = ! empty( $assoc_args['start-date'] ) ? strtotime( $assoc_args['start-date'] ) : strtotime( '-90 days', $now );
+
+		\WP_CLI::log(
+			'
+Fetching active subscriptions with missing or missed next_payment dates...
+		'
+		);
+
+		$query_args    = [
+			'subscriptions_per_page' => $batch_size,
+			'subscription_status'    => [ 'active', 'pending' ],
+			'offset'                 => 0,
+		];
+		$processed     = 0;
+		$subscriptions = \wcs_get_subscriptions( $query_args );
```

Comment:
On larger sites, this might take a while – it's querying for _all_ active & pending subs. Direct DB query will likely be way more performant. It has to take into account that some sites use HPOS, while some don't, though.

Reply:
It only queries for `$batch_size` (50 by default) at a time, and stops as soon as it finds a subscription with a creation date older than the `--start-date` date (90 days ago by default). So it'll take a while if you give it a start date far in the past and/or there are a ton of subscriptions since the start date, but I think for most sites this shouldn't be much of an issue.

Reply:
Yeah, got it. Should be fine, we'll see on these bigger sites 🤞